crates/ruff/src/rules/pydocstyle/rules/sections.rs

```diff
@@ -923,7 +923,7 @@ fn parameters_section(checker: &mut Checker, docstring: &Docstring, context: &Se
 
     // Join line continuations, then resplit by line.
     let adjusted_following_lines = context.following_lines.join("\n").replace("\\\n", "");
-    let lines: Vec<&str> = LinesWithTrailingNewline::from(&adjusted_following_lines).collect();
+    let lines: Vec<&str> = NewlineWithTrailingNewline::from(&adjusted_following_lines).collect();
```

Comment:
Nit: This can be implemented without the need for `collect`

```rust
    let mut lines = NewlineWithTrailingNewline::from(&adjusted_following_lines);

    if let Some(mut current_line) = lines.next() {
        for next_line in lines {
            let current_leading_space = whitespace::leading_space(current_line);
            if current_leading_space == section_level_indent
                && (whitespace::leading_space(next_line).len() > current_leading_space.len())
                && !next_line.trim().is_empty()
            {
                let parameters = if let Some(semi_index) = current_line.find(':') {
                    // If the parameter has a type annotation, exclude it.
                    &current_line[..semi_index]
                } else {
                    // Otherwise, it's just a list of parameters on the current line.
                    current_line.trim()
                };
                // Notably, NumPy lets you put multiple parameters of the same type on the same
                // line.
                for parameter in parameters.split(',') {
                    docstring_args.insert(parameter.trim());
                }
            }

            current_line = next_line;
        }
    }
```